api/provider/aws/test_aws/load_balancer_create_test.go

```diff
@@ -0,0 +1,275 @@
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+	mockConfig.EXPECT().VPC().Return("vpc_id").AnyTimes()
+	mockConfig.EXPECT().Region().Return("region").AnyTimes()
+	mockConfig.EXPECT().AccountID().Return("123456789012")
+	mockConfig.EXPECT().PrivateSubnets().Return([]string{"priv1", "priv2"}).AnyTimes()
+	mockConfig.EXPECT().PublicSubnets().Return([]string{"pub1", "pub2"}).AnyTimes()
+
+	defer provider.SetEntityIDGenerator("lb_id")()
+
+	req := models.CreateLoadBalancerRequest{
+		LoadBalancerName: "lb_name",
+		EnvironmentID:    "env_id",
+		IsPublic:         true,
+		Ports: []models.Port{
+			{
+				CertificateName: "cert",
+				ContainerPort:   88,
+				HostPort:        80,
+				Protocol:        "http",
+			},
+			{
+				CertificateName: "cert",
+				ContainerPort:   4444,
+				HostPort:        443,
+				Protocol:        "https",
+			},
+		},
+		HealthCheck: models.HealthCheck{
+			Target:             "HTTPS:443/path/to/site",
+			Interval:           20,
+			Timeout:            15,
+			HealthyThreshold:   4,
+			UnhealthyThreshold: 3,
+		},
+	}
+
+	readSGHelper(mockAWS, "l0-test-env_id-env", "env_sg")
+	createSGHelper(t, mockAWS, "l0-test-lb_id-lb", "vpc_id")
+	readSGHelper(mockAWS, "l0-test-lb_id-lb", "lb_sg")
+
+	ingressInput1 := &ec2.AuthorizeSecurityGroupIngressInput{}
+	ingressInput1.SetGroupId("lb_sg")
+	ingressInput1.SetCidrIp("0.0.0.0/0")
+	ingressInput1.SetIpProtocol("TCP")
+	ingressInput1.SetFromPort(int64(80))
+	ingressInput1.SetToPort(int64(80))
+
+	mockAWS.EC2.EXPECT().
+		AuthorizeSecurityGroupIngress(ingressInput1).
+		Return(&ec2.AuthorizeSecurityGroupIngressOutput{}, nil)
+
+	ingressInput2 := &ec2.AuthorizeSecurityGroupIngressInput{}
+	ingressInput2.SetGroupId("lb_sg")
+	ingressInput2.SetCidrIp("0.0.0.0/0")
+	ingressInput2.SetIpProtocol("TCP")
+	ingressInput2.SetFromPort(int64(443))
+	ingressInput2.SetToPort(int64(443))
+
+	mockAWS.EC2.EXPECT().
+		AuthorizeSecurityGroupIngress(ingressInput2).
+		Return(&ec2.AuthorizeSecurityGroupIngressOutput{}, nil)
+
+	iamRoleInput := &iam.CreateRoleInput{}
+	iamRoleInput.SetRoleName("l0-test-lb_id-lb")
+	iamRoleInput.SetAssumeRolePolicyDocument(provider.DefaultAssumeRolePolicy)
+
+	mockAWS.IAM.EXPECT().
+		CreateRole(iamRoleInput).
+		Return(&iam.CreateRoleOutput{}, nil)
+
+	renderedLBRolePolicy, err := provider.RenderLoadBalancerRolePolicy(
+		"region",
+		"123456789012",
+		"l0-test-lb_id",
+		provider.DefaultLBRolePolicyTemplate)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	putIAMPolicyInput := &iam.PutRolePolicyInput{}
+	putIAMPolicyInput.SetPolicyName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetRoleName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetPolicyDocument(renderedLBRolePolicy)
+
+	mockAWS.IAM.EXPECT().
+		PutRolePolicy(putIAMPolicyInput).
+		Return(&iam.PutRolePolicyOutput{}, nil)
+
+	serverCertificateMetadata := &iam.ServerCertificateMetadata{}
+	serverCertificateMetadata.SetArn("cert")
```

Comment:
the cert arn and name need to be different values; we need to ensure proper use of name vs. arn elsewhere